internal/backend/windows/wslexe_windows.go

```diff
@@ -45,3 +50,56 @@ func (Backend) SetAsDefault(distroName string) error {
 	}
 	return nil
 }
+
+// State returns the state of a particular distro as seen in `wsl.exe -l -v`.
+func (Backend) State(distributionName string) (s state.State, err error) {
+	cmd := exec.Command("wsl.exe", "--list", "--all", "--verbose")
+	cmd.Env = append(cmd.Env, "WSL_UTF8=1")
+
+	out, err := cmd.Output()
+	if err != nil {
+		return s, err
+	}
+
+	/*
+		Sample output:
+		   NAME           STATE           VERSION
+		 * Ubuntu         Stopped         2
+		   Ubuntu-Preview Running         2
+	*/
+
+	sc := bufio.NewScanner(bytes.NewReader(out))
+
+	// Discarding header
+	sc.Scan()
+	if err := sc.Err(); err != nil {
+		return s, err
+	}
+
+	for sc.Scan() {
+		data := strings.Fields(sc.Text())
+		if len(data) == 4 {
+			// default distro, ignoring leading asterisk
+			data = data[1:]
+		}
+
+		if data[0] == distributionName {
+			return parseState(data[1])
+		}
+	}
+
+	return state.NonRegistered, nil
+}
+
+func parseState(s string) (state.State, error) {
```

Comment:
Shouldn’t that function be in the state package?

Like `NewFromString(s string) (state.State, error)`

(ofc, then, you need a test in state package :)

Reply:
Fair enough. I moved it and wrote the tests.